### PR TITLE
Overrides for perl_cpan dependency manager

### PR DESCRIPTION
**Pull request by sersut**
This add support for overrides to cpan dep manager. Also we add the required overrides to be able to run licensing for Chef Server correctly.

/cc: @danielsdeleo @ryancragun

### CODE REVIEW

File: spec/license_scout/dependency_manager/cpan_spec.rb [hunk 1]
```diff
@@ -290,19 +290,70 @@ def fixture_pkg(pkg_name)
         expect(capture_tiny.license_files.first).to eq(expected_path)
 
         libintl = cpan.dependencies.find { |d| d.name == "libintl-perl" }
-        expect(libintl.license).to be_nil
+        expect(libintl.license).to eq("GPL-3.0") # from default overrides
```
Comment (danielsdeleo) [inline spec/license_scout/dependency_manager/cpan_spec.rb]: seems like we should still have one test case where the license and license file are not found and there isn't an override.

**Comment by danielsdeleo** [comment]
one nitpick about the tests, but 👍

File: spec/license_scout/dependency_manager/cpan_spec.rb (hunk 1 above)
Comment (sersut) [inline spec/license_scout/dependency_manager/cpan_spec.rb]: Will add one.

Comment (sersut) [inline spec/license_scout/dependency_manager/cpan_spec.rb]: Actually @danielsdeleo looks like `PathTools` is the case that covers this.

**Pull request merged**